osbuild/__main__.py

```diff
@@ -10,6 +10,25 @@
 RED = "\033[31m"
 
 
+def mark_checkpoints(pipeline, checkpoints):
+    points = set(checkpoints)
+    def mark_stage(stage):
+        for c in points:
+            if stage.id == c:
+                stage.snapshot = True
```

Comment:
This might be an oversight from search/replacing from a previous iteration? `pipeline.py` looks for `stage.checkpoint`, not `stage.snapshot`.

Changing this makes it work (yay, loving it!), but there seem to be other mentions of `snapshot` still.

Reply:
Worked on that during a talk at FOSDEM and was obviously distracted. Sorry about that.